storage/innobase/fsp/fsp0fsp.cc

```diff
@@ -3109,10 +3109,1760 @@ buf_block_t *fsp_get_latched_xdes_page(
     BUF_GET_POSSIBLY_FREED, mtr, err);
 }
 
+class spaceDefragmenter;
+
+namespace flst
+{
+  /** Validate the file list node for the system tablespace.
+  @param addr file space address
+  @return true if validation successful or false */
+  static bool node_valid(const fil_addr_t *addr) noexcept
```

Comment:
All the functions in the `namespace` need to be declared `static`, because they are not to be called from other compilation units.